WordPress/Classes/Models/Blog+Editor.swift

```diff
@@ -0,0 +1,34 @@
+import Foundation
+
+enum MobileEditor: String {
+    case aztec
+    case gutenberg
+}
+
+enum WebEditor: String {
+    case classic
+    case gutenberg
+}
+
+extension Blog {
+    struct Editor {
+        fileprivate let blog: Blog
+        var mobile: MobileEditor? {
+            return MobileEditor(rawValue: blog.mobileEditor ?? "")
+        }
+        var web: WebEditor? {
+            return WebEditor(rawValue: blog.webEditor ?? "")
+        }
+        func setMobileEditor(_ newValue: MobileEditor) {
+            blog.mobileEditor = newValue.rawValue
+        }
+    }
+
+    var editor: Editor {
+        return Editor(blog: self)
```

Comment:
Sorry if this is a stupid question, but would this be a reference cycle?

Reply:
I don't think so? 🤔 
Blog.Editor is a Value type. It does hold a reference to a Blog object, but then Blog just construct an Editor once and gives it back to the caller, and its lifetime will die together with the stack frame, so the ref count for Blog should go down accordingly.

I did a small test just to be sure and I didn't notice any Blog object leaking.

Please let me know if I'm missing something!